src/React/Renderers/VtkGeometryRenderer/index.js

```diff
@@ -0,0 +1,270 @@
+/* global window */
+
+import React                           from 'react';
+
+import vtkOpenGLRenderWindow           from 'vtk.js/Sources/Rendering/OpenGL/RenderWindow';
+import vtkSynchronizableRenderWindow   from 'vtk.js/Sources/Rendering/Misc/SynchronizableRenderWindow';
+import vtkRenderWindowInteractor       from 'vtk.js/Sources/Rendering/Core/RenderWindowInteractor';
+import vtkInteractorStyleManipulator   from 'vtk.js/Sources/Interaction/Style/InteractorStyleManipulator';
+import vtkTrackballPan                 from 'vtk.js/Sources/Interaction/Manipulators/TrackballPan';
+import vtkTrackballZoom                from 'vtk.js/Sources/Interaction/Manipulators/TrackballZoom';
+import vtkTrackballRotate              from 'vtk.js/Sources/Interaction/Manipulators/TrackballRotate';
+
+
+const SYNCHRONIZATION_CONTEXT_NAME = 'pvwLocalRenderingContext';
+const ACTIVE_VIEW_ID = '-1';
+
+export default class VtkGeometryRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.geometryTopicSubscription = null;
+    this.openGlRenderWindow = null;
+    this.renderWindow = null;
+    this.interactorStyle = null;
+    this.synchCtx = null;
+
+    // Set up initial state
+    this.state = {
+      viewId: props.viewId,
+    };
+
+    this.viewChanged = this.viewChanged.bind(this);
+    this.addViewObserver = this.addViewObserver.bind(this);
+    this.removeViewObserver = this.removeViewObserver.bind(this);
+    this.subscribeViewChangeTopic = this.subscribeViewChangeTopic.bind(this);
+    this.unsubscribeViewChangeTopic = this.unsubscribeViewChangeTopic.bind(this);
+    this.updateRenderWindowSize = this.updateRenderWindowSize.bind(this);
+  }
+
+  componentDidMount() {
+    // Get our hands on the default synchronization context and tell how to fetch data arrays
+    this.synchCtx = vtkSynchronizableRenderWindow.getSynchronizerContext(this.props.synchronizerContextName);
+    this.synchCtx.setFetchArrayFunction(this.props.client.VtkGeometryDelivery.getArray);
```

Comment:
Could some of those initialization move to the constructor? (Because they don't need the DOM)

This way `componentDidMount` just show the DOM relationship.